lib/features/feature_example/presentation/screens/api_interact_example/api_interact_example_model.dart

```diff
@@ -0,0 +1,43 @@
+import 'package:elementary/elementary.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter_template/core/architecture/domain/entity/result.dart';
+import 'package:flutter_template/features/feature_example/domain/repositories/feature_example_repository.dart';
+import 'package:flutter_template/features/feature_example/presentation/screens/api_interact_example/api_interact_example_screen.dart';
+import 'package:flutter_template/features/feature_example/presentation/screens/api_interact_example/api_interact_example_state.dart';
+
+/// {@template feature_example_model.class}
+/// [ElementaryModel] for [ApiInteractExampleScreen]
+/// {@endtemplate}
+final class ApiInteractExampleModel extends ElementaryModel {
+  final IFeatureExampleRepository _repository;
+
+  /// {@macro feature_example_model.class}
+  ApiInteractExampleModel({
+    required IFeatureExampleRepository repository,
+    super.errorHandler,
+  }) : _repository = repository;
+
+  final _ipState = ValueNotifier<ApiInteractExampleState>(ApiInteractExampleStateInitial());
+
+  /// State of screen.
+  ValueListenable<ApiInteractExampleState> get screenState => _ipState;
+
+  /// Load IP address.
+  Future<void> loadIp() async {
+    _emit(ApiInteractExampleStateLoading());
+
+    final result = await _repository.getIp();
+
+    switch (result) {
+      case ResultOk(:final data):
+        _emit(ApiInteractExampleStateLoaded(data.ip));
+      case ResultFailed():
+        _emit(ApiInteractExampleStateError());
+    }
+  }
+
+  // ignore: use_setters_to_change_properties
+  void _emit(ApiInteractExampleState state) {
+    _ipState.value = state;
+  }
```

Comment:
We should think how not to ignore this rule, maybe we can somehow change the code.

Reply:
If I convert this method to a setter, it will require a getter that will not be used anywhere. Maybe we should ignore this rule?

<img width="735" alt="Screenshot 2024-02-29 at 11 42 28" src="https://github.com/surfstudio/surf-flutter-app-template/assets/63193897/8a0e702b-376f-4a10-a106-05788904df75">

Reply:
I like the _emit method as it makes states more clear, but I think that ignoring a rule is a bad idea.
I suggest to make the following extension for the `ValueNotifier`:
```dart
extension ValueNotifierX<T> on ValueNotifier<T> {
  void emit(T value) {
    this.value = value;
  }
}
```
Then we can use it like this: 
```dart 
_state.emit(newState);
```